cartographer/mapping/id.h

```diff
@@ -322,6 +323,43 @@ class MapById {
 
   bool empty() const { return begin() == end(); }
 
+  ConstIterator lower_bound(int trajectory_id, const common::Time& time) {
+    if (SizeOfTrajectoryOrZero(trajectory_id) == 0) {
+      return EndOfTrajectory(trajectory_id);
+    }
+
+    const std::map<int, DataType>& trajectory =
+        trajectories_.at(trajectory_id).data_;
+    if (std::prev(trajectory.end())->second.time() < time) {
```

Comment:
CHECK(!trajectory.empty())? Or is this an invariant that you guarantee?

Reply:
This is guaranteed by 'SizeOfTrajectoryOrZero(trajectory_id) == 0'.